File: string/check-if-all-characters-have-equal-number-of-occurrences/check-if-all-characters-have-equal-number-of-occurrences.go
package checkifallcharactershaveequalnumberofoccurrences

func areOccurrencesEqual(s string) bool {
	count := [27]int{}

	for i := 0; i < len(s); i++ {
		count[s[i]-'a']++
	}

	f := count[s[0]-'a']

	for _, v := range count {
		if v != 0 && v != f {
			return false
		}
	}

	return true
}

func areOccurrencesEqualMoreSpace(s string) bool {
	count := make(map[rune]int)
	var f int

	for _, c := range s {
		if _, ok := count[c]; ok {
			count[c]++
		} else {
			count[c] = 1
		}
		f = count[c]
	}

	for _, v := range count {
		if v != f {
			return false
		}
	}

	return true
}

File: string/check-if-all-characters-have-equal-number-of-occurrences/check-if-all-characters-have-equal-number-of-occurrences.ts
const A = "a".charCodeAt(0);

function areOccurrencesEqual(s: string): boolean {
    const lookup = new Array(27).fill(0);

    for (let c of s) {
        lookup[c.charCodeAt(0) - A]++;
    }
    const count = lookup[s[0].charCodeAt(0) - A];

    for (let i = 0; i < lookup.length; i++) {
        if (lookup[i] !== 0 && lookup[i] !== count) {
            return false;
        }
    }

    return true;
};
